src/applications/yellow-ribbon/containers/SearchResults/index.jsx

```diff
@@ -0,0 +1,143 @@
+// Dependencies.
+import React, { Component } from 'react';
+import AlertBox from '@department-of-veterans-affairs/formation-react/AlertBox';
+import LoadingIndicator from '@department-of-veterans-affairs/formation-react/LoadingIndicator';
+import Pagination from '@department-of-veterans-affairs/formation-react/Pagination';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { map } from 'lodash';
+// Relative imports.
+import SearchResult from '../../components/SearchResult';
+import { fetchResultsThunk, updatePageAction } from '../../actions';
+
+export class SearchResults extends Component {
+  static propTypes = {
+    // From mapStateToProps.
+    error: PropTypes.string.isRequired,
+    fetching: PropTypes.bool.isRequired,
+    results: PropTypes.arrayOf(
+      PropTypes.shape({
+        // Original form data key-value pairs.
+        name: PropTypes.string.isRequired,
+        id: PropTypes.string.isRequired,
+        city: PropTypes.string.isRequired,
+        state: PropTypes.string.isRequired,
+      }).isRequired,
+    ),
+    page: PropTypes.number.isRequired,
+    perPage: PropTypes.number.isRequired,
+    totalResults: PropTypes.number,
+  };
+
+  onPageSelect = page => {
+    const { fetchResults, perPage, updatePage } = this.props;
+
+    // Derive the current name params.
+    const queryParams = new URLSearchParams(window.location.search);
+
+    // Derive the state values from our query params.
+    const name = queryParams.get('name') || '';
+    const state = queryParams.get('state') || '';
+
+    // Update the page.
+    updatePage(page);
+
+    // Refetch results.
+    fetchResults({ page, perPage, name, hideFetchingState: true, state });
+
+    // Scroll to top.
+    window.scrollTo({ behavior: 'smooth', top: 0 });
+  };
+
+  render() {
+    const { onPageSelect } = this;
+    const {
+      error,
+      fetching,
+      page,
+      perPage,
+      results,
+      totalResults,
+    } = this.props;
+
+    // Show loading indicator if we are fetching.
+    if (fetching) {
+      return <LoadingIndicator message="Loading search results..." />;
+    }
+
+    // Show the error alert box if there was an error.
+    if (error) {
+      return (
+        <AlertBox
+          headline="Something went wrong"
+          content={error}
+          status="error"
+        />
+      );
+    }
+
+    // Do not render if we have not fetched, yet.
+    if (!results) {
+      return null;
+    }
+
+    // Show no results found message.
+    if (!results.length) {
+      return (
+        <h2 className="vads-u-font-size--lg vads-u-margin-top--1p5 vads-u-font-weight--normal">
+          No results found.
+        </h2>
+      );
+    }
+
+    // Derive values for "Displayed x-x out of x results."
+    const resultsStartNumber = page * perPage - (perPage - 1);
+    const resultsEndNumber =
+      page * perPage > totalResults ? totalResults : page * perPage;
```

Comment:
I wish this logic could be simpler. If you have any concerns about edge cases or want to have dedicated tests, I'm kinda wondering if it's worth abstracting this logic into a local `Pagination` component, passing this stuff as props, doing the calculations, then passing it to the design system components.

Reply:
I just wrote some abstracted functions to make it (hopefully) much more readable! 🙂